components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/mappings/SubscriptionMappingUtil.java

```diff
@@ -0,0 +1,210 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.publisher.v1.utils.mappings;
+
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.APIProvider;
+import org.wso2.carbon.apimgt.api.dto.UserApplicationAPIUsage;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.apimgt.api.model.Application;
+import org.wso2.carbon.apimgt.api.model.SubscribedAPI;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.APIInfoDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.ApplicationInfoDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.ExtendedSubscriptionDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.PaginationDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.SubscriptionDTO;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.SubscriptionListDTO;
+import org.wso2.carbon.apimgt.rest.api.util.RestApiConstants;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is responsible for mapping APIM core subscription related objects into REST API subscription related DTOs
+ */
+public class SubscriptionMappingUtil {
+
+    /**
+     * Converts a SubscribedAPI object into SubscriptionDTO
+     *
+     * @param subscription SubscribedAPI object
+     * @return SubscriptionDTO corresponds to SubscribedAPI object
+     */
+    public static SubscriptionDTO fromSubscriptionToDTO(SubscribedAPI subscription)
+            throws APIManagementException {
+        SubscriptionDTO subscriptionDTO = new SubscriptionDTO();
+        subscriptionDTO.setSubscriptionId(subscription.getUUID());
+        subscriptionDTO.setApiInfo(FromAPIIdToAPIinfoDTO(subscription.getApiId()));
```

Comment:
shall we remove this line?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/6772